sdks/java/core/src/main/java/org/apache/beam/sdk/values/reflect/RecordFactory.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.beam.sdk.values.reflect;
+
+import com.google.common.collect.ImmutableList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import net.bytebuddy.ByteBuddy;
+import org.apache.beam.sdk.values.BeamRecord;
+import org.apache.beam.sdk.values.BeamRecordType;
+import org.apache.beam.sdk.values.reflect.field.DirectAccessGetterFactory;
+import org.apache.beam.sdk.values.reflect.field.FieldValueGetter;
+import org.apache.beam.sdk.values.reflect.field.GetterMethodGetterFactory;
+
+/**
+ * Generates the code to create BeamRecordTypes and BeamRecords based on pojos.
+ *
+ * <p>Generated record types are cached in the instance of this factory.
+ *
+ * <p>Currently only 1-1 mapping is supported between pojo classes and record types.
+ */
+public class RecordFactory {
```

Comment:
<!--thread_id:cc_154609288_t; commit:12389cee6a5061c472533ec063d041e03007687b; resolved:1-->
<!--section:context-quote-->
> **iemejia** wrote:
> BeamRecordFactory ? Probably to remind us in the future to rename this when we move BeamRecord -> Row (or Record).

<!--section:body-->
Sure, renamed this one as it is the main interface. Kept other classes without `Beam` prefix for now. Let me know if you think other ones would be also better with `Beam`